### PR TITLE
Don't start new tests if the Jenkins queue is full.

### PR DESCRIPTION
**Pull request by spxtr**
This isn't the best way to do this, but it'll have to do for now.

### CODE REVIEW

File: prow/cmd/line/main.go [hunk 1]
```diff
@@ -62,7 +62,10 @@ var (
 	jenkinsTokenFile = flag.String("jenkins-token-file", "/etc/jenkins/jenkins", "Path to the file containing the Jenkins API token.")
 )
 
-const guberBase = "https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull"
+const (
+	guberBase = "https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull"
+	testInfra = "https://github.com/kubernetes/test-infra/issues"
```
Comment (krzyzacy) [inline prow/cmd/line/main.go]: eh, I feel like it somehow just create an issue or send a ping to us

File: prow/cmd/line/main.go [hunk 2]
```diff
@@ -257,7 +260,15 @@ func (c *testClient) TestPRKubernetes() error {
 // TestPRJenkins starts a Jenkins build and watches it, updating the GitHub
 // status as necessary.
 func (c *testClient) TestPRJenkins() error {
+	if size, err := c.JenkinsClient.QueueSize(); err != nil {
+		c.tryCreateStatus(github.StatusError, "Error checking Jenkins queue.", testInfra)
+		return err
+	} else if size > 200 {
+		c.tryCreateStatus(github.StatusError, "Jenkins overloaded. Please try again later.", testInfra)
```
Comment (fejta) [inline prow/cmd/line/main.go]: Maybe specify that there are more than 200 pending jobs?

File: prow/cmd/line/main.go [hunk 3]
```diff
@@ -257,7 +260,15 @@ func (c *testClient) TestPRKubernetes() error {
 // TestPRJenkins starts a Jenkins build and watches it, updating the GitHub
 // status as necessary.
 func (c *testClient) TestPRJenkins() error {
+	if size, err := c.JenkinsClient.QueueSize(); err != nil {
```
Comment (rmmh) [inline prow/cmd/line/main.go]: it seems bad that a (transient?) 500 getting the queue size mean that you won't even try to start the build

Comment (spxtr) [inline prow/cmd/line/main.go]: I retry 500s.

**Pull request merged**